modules/wordads/php/widgets.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/**
+ * Widget for inserting an ad into your sidebar
+ *
+ * @since 0.1
+ * TODO: sanitize all the output
+ */
+class WordAds_Sidebar_Widget extends WP_Widget {
+
+	private static $allowed_tags = array( 'mrec', 'wideskyscraper' );
+
+	private $options = array();
+
+	function __construct() {
+		parent::__construct(
+			'wordads_sidebar_widget',
+			__( 'WordAds Widget (Jetpack)', 'jetpack' ),
+			array( 'description' => __( 'Insert a WordAds ad wherever you can place a widget.', 'jetpack' ) )
+		);
+
+		$this->options = get_option( 'wordads_settings',  array() );
+	}
+
+	/**
+	 * Convenience function for grabbing options from params->options
+	 * @param  string $option the option to grab
+	 * @param  mixed  $default (optional)
+	 * @return option or $default if not set
+	 *
+	 * @since 0.1
+	 */
+	function option( $option, $default = false ) {
+		if ( ! isset( $this->options[$option] ) )
+			return $default;
+
+		return $this->options[$option];
+	}
+
+	public function widget( $args, $instance ) {
+		global $wordads;
+		if ( $wordads->should_bail() ) {
+			return false;
+		}
+
+		$about = __( 'About these ads', 'jetpack' );
+		$section_id = 0 === $wordads->params->blog_id ? WORDADS_API_TEST_ID : $wordads->params->blog_id . '3';
+		$width = WordAds::$ad_tag_ids[$instance['unit']]['width'];
+		$height = WordAds::$ad_tag_ids[$instance['unit']]['height'];
+		$data_tags = ( $wordads->params->cloudflare ) ? ' data-cfasync="false"' : '';
+		echo <<< HTML
+		<div class="wpcnt">
+			<div class="wpa">
+				<a class="wpa-about" href="http://en.wordpress.com/about-these-ads/" rel="nofollow">$about</a>
```

Comment:
Maybe we should create a new page specific to Jetpack sites? The current support doc says "The site you just visited is part of WordPress.com. ", so it could be a bit confusing.